app/models/manageiq/providers/ibm_terraform/configuration_manager.rb

```diff
@@ -84,4 +59,8 @@ def provider
   def ensure_provider
     build_provider.tap { |p| p.configuration_manager = self }
   end
+
+  def ensure_zone_id
+    self.zone_id ||= provider.zone_id
```

Comment:
Since the zone_id is being set by the create_from_params method on the ConfigurationManager I think we want to be setting the provider.zone_id from self.zone_id instead of the other way around.